nix/pkgs/docker-image/default.nix

```
@@ -5,6 +5,20 @@ let
 
     set -eu
 
+    # set defaults
+    amesPort=${toString amesPort}
+
+    # check args
+    for i in "$@"
+    do
+    case $i in
+        -p=*|--port=*)
+            amesPort="${i#*=}"
```

The error is actually here,  you need to escape the nix antiquotation by prefixing `$` with two single quotes per
https://nixos.org/manual/nix/stable/#idm140737322057376
@wexpertsystems @joemfb 
```suggestion
            amesPort="''${i#*=}"
```